File: interview questions/findIntersection.py
# Input = ["1, 3, 4, 7, 13", "1, 2, 4, 13, 15"]
# Output = "1,4,13"

def find_intersection(strArr):
    results = []
    arr1 = strArr[0].split(", ")
    arr2 = strArr[1].split(", ")

    for i in range(len(arr1)):
        if arr1[i] in arr2:
            results.append(arr1[i])

    return ",".join(results) if len(results) > 0 else false


# strArr = ["1, 3, 4, 7, 13", "1, 2, 4, 13, 15"]
# print find_intersection(strArr)

File: interview questions/findIntersection.js
// Input = ["1, 3, 4, 7, 13", "1, 2, 4, 13, 15"]
// Output = "1,4,13"

function FindIntersection(strArr) {
  const results = [];
  const arr1 = strArr[0].split(", ");
  const arr2 = strArr[1].split(", ");

  for (const num1 of arr1) {
    if (arr2.includes(num1)) results.push(num1);
  }

  return results.length ? results.join(",") : false;
}
